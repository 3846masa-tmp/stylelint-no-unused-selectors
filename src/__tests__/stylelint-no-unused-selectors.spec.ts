import path from 'path';
import stylelint from 'stylelint';

const fixturesRoot = path.join(__dirname, '..', '..', 'examples');
const configFilePath = path.join(fixturesRoot, '.stylelintrc.js');

function parseResult(result: stylelint.LinterResult): unknown {
  const parsedOutputs = JSON.parse(result.output);

  if (parsedOutputs.length > 1) {
    throw Error('More than one result received');
  }

  const output = parsedOutputs[0];
  const { source: _, ...filteredOutput } = output;

  return filteredOutput;
}

test('Disabllow selectors that are not used in a HTML file', async (): Promise<
  void
> => {
  const options = {
    configFile: configFilePath,
    files: path.join(fixturesRoot, 'html', '*.css'),
  };

  const result = await stylelint.lint(options);

  expect(result.errored).toBe(true);
  expect(parseResult(result)).toMatchSnapshot();
});

test('Disabllow selectors that are not used in a JSX file', async (): Promise<
  void
> => {
  const options = {
    configFile: configFilePath,
    files: path.join(fixturesRoot, 'jsx', '*.css'),
  };

  const result = await stylelint.lint(options);

  expect(result.errored).toBe(true);
  expect(parseResult(result)).toMatchSnapshot();
});

test('Disabllow selectors that are not used in a JSX file using CSS Modules', async (): Promise<
  void
> => {
  const options = {
    configFile: configFilePath,
    files: path.join(fixturesRoot, 'jsx-with-css-modules', '*.css'),
  };

  const result = await stylelint.lint(options);

  expect(result.errored).toBe(true);
  expect(parseResult(result)).toMatchSnapshot();
});
