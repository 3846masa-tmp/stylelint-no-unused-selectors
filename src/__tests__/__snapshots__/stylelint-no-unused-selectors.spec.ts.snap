// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabllow selectors that are not used in a HTML file 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 27,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.html. (plugin/no-unused-selectors)",
    },
    Object {
      "column": 1,
      "line": 32,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".list-item:last-child + li is defined but doesn't match any elements in test.html. (plugin/no-unused-selectors)",
    },
  ],
}
`;

exports[`Disabllow selectors that are not used in a JSX file 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 19,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.jsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;

exports[`Disabllow selectors that are not used in a JSX file using CSS Modules 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 23,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.jsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;

exports[`Disabllow selectors that are not used in a TSX file 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 13,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.tsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;

exports[`Disabllow selectors that are not used in a TSX file using CSS Modules 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 23,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.tsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;

exports[`Disabllow selectors that are not used in a flow-typed JSX file 1`] = `
Object {
  "deprecations": Array [],
  "errored": true,
  "invalidOptionWarnings": Array [],
  "parseErrors": Array [],
  "warnings": Array [
    Object {
      "column": 1,
      "line": 13,
      "rule": "plugin/no-unused-selectors",
      "severity": "error",
      "text": ".not-found is defined but doesn't match any elements in test.jsx. (plugin/no-unused-selectors)",
    },
  ],
}
`;
